realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -165,7 +165,11 @@ static inline size_t do_create_row(jlong shared_realm_ptr, jlong table_ptr)
     auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
     auto& table = *(reinterpret_cast<realm::Table*>(table_ptr));
     shared_realm->verify_in_write();
+#if REALM_ENABLE_SYNC
+    return sync::create_object(shared_realm->read_group(), table);
```

Comment:
Didn't OS add some feature flag so the same API could be used for both 1.* and 2.* or did I misunderstood something?